build.csx

```diff
@@ -0,0 +1,102 @@
+#r "System.IO.Compression.FileSystem.dll"
+#load "src/packages/simple-targets-csharp.2.0.0/simple-targets-csharp.csx"
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.IO.Compression;
+using System.Net;
+
+// locations
+var resharperCltUrl = "https://download.jetbrains.com/resharper/JetBrains.ReSharper.CommandLineTools.2016.2.20160912.114811.zip";
+var resharperCltPath = $"{Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData)}/.resharper/JetBrains.ReSharper.CommandLineTools.2016.2.20160912.114811.zip";
+var inspectCodePath = "./.resharper/inspectcode.exe";
+var msBuild = $"{Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86)}/MSBuild/14.0/Bin/msbuild.exe";
+var solution = "./src/NServiceBus.RabbitMq.sln";
+var dotSettings = "./src/NServiceBus.RabbitMQ.sln.DotSettings";
+var nunit = "./src/packages/NUnit.ConsoleRunner.3.4.1/tools/nunit3-console.exe";
+var unitTests = "./src/NServiceBus.RabbitMQ.Tests/bin/Release/NServiceBus.Transports.RabbitMQ.Tests.dll";
+var acceptanceTests = "./src/NServiceBus.RabbitMQ.AcceptanceTests/bin/Release/NServiceBus.RabbitMQ.AcceptanceTests.dll";
+var transportTests = "./src/NServiceBus.Transport.RabbitMQ.TransportTests/bin/Release/NServiceBus.Transport.RabbitMQ.TransportTests.dll";
+
+// targets
+var targets = new Dictionary<string, Target>();
+
+targets.Add("default", new Target { DependOn = new[] { "build", "inspect", "unit-test", "acceptance-test", "transport-test" } });
+
+targets.Add("build", new Target { Do = () => Cmd(msBuild, $"{solution} /p:Configuration=Release /nologo /m /v:m /nr:false"), });
+
+targets.Add(
+    "download-resharper-clt",
+    new Target { Do = () => { if (!File.Exists(resharperCltPath)) Download(resharperCltUrl, resharperCltPath); }, });
+
+targets.Add(
+    "unzip-resharper-clt",
+    new Target
+    {
+        DependOn = new[] { "download-resharper-clt" },
+        Do = () => { if (!File.Exists(inspectCodePath)) ZipFile.ExtractToDirectory(resharperCltPath, Path.GetDirectoryName(inspectCodePath)); },
+    });
+
+targets.Add(
+    "inspect",
+    new Target
+    {
+        DependOn = new[] { "build", "unzip-resharper-clt", },
+        Do = () => Cmd(inspectCodePath, $"--profile={dotSettings} {solution}"),
```

Comment:
It turns out that this doesn't have the desired effect. The exit code is always zero regardless of which types of violations are found is to output the reports to disk and then inspect those which are marked with "error" to see if there is anything which will fail on the CI server. I'll continue to investigate, but it seems like we may have to write some code to do that, since it is taken care of by something built into TeamCity when run on the CI server.